common/test/acceptance/pages/lms/problem.py

```diff
@@ -147,6 +146,27 @@ def is_focus_on_problem_meta(self):
         """
         return self.q(css='.problem-header').focused
 
+    def is_submit_disabled(self):
+        """
+        Checks if the submit button is disabled
+        """
+        return (self.q(css='.problem .submit').attrs('disabled') and
+                'is-disabled' in self.q(css='.problem .submit').attrs('class')[0])
+
+    def is_focus_on_submit_notification(self):
+        """
+        Check for focus submit notification.
+        """
+
+        def focus_check():
+            """
+            Checks whether or not the focus is on the notification-submit
+            """
+            return self.q(css='.notification-submit').focused
+
+        self.wait_for(promise_check_func=focus_check, description='Waiting for the notification-submit to gain focus')
+        return focus_check()
```

Comment:
Another thought-- maybe the focus check should just move into the wait_success_notification_visible, wait_incorrect_notification_visible, and wait_partial_notification_visible helper methods. It looks like you always want to check that focus is there after checking visibility.